src/main/java/de/dennisguse/opentracks/viewmodels/IntervalStatistics.java

```diff
@@ -111,5 +144,23 @@ public float getGain_m() {
         public float getLoss_m() {
             return loss_m;
         }
+
+        private void update(TrackStatistics trackStatistics, @Nullable TrackPoint lastTrackPoint) {
+            distance = distance.plus(trackStatistics.getTotalDistance());
+            time = time.plus(trackStatistics.getTotalTime());
+            gain_m = trackStatistics.hasTotalAltitudeGain() ? trackStatistics.getTotalAltitudeGain() : gain_m;
+            loss_m = trackStatistics.hasTotalAltitudeLoss() ? trackStatistics.getTotalAltitudeLoss() : loss_m;
+            if (lastTrackPoint != null) {
+                gain_m = lastTrackPoint.hasAltitudeGain() ? gain_m - lastTrackPoint.getAltitudeGain() : gain_m;
```

Comment:
should that not be part of the trackstatistics already?
Or was the lastTrackPoint not yet added?

Reply:
That's because every time a new Interval is added trackStatisticsUpdater is reset to begin from the last track point, so last track point is added two times, so we need to substract the gain and the loss in that case.